src/packages/emmett-mongodb/src/eventStore/storage/mongoDBEventStoreStorage.unit.spec.ts

```diff
@@ -0,0 +1,432 @@
+import { assertEqual, assertNotEqual } from '@event-driven-io/emmett';
+import type { MongoClient } from 'mongodb';
+import { beforeEach, describe, it } from 'node:test';
+import { v7 as uuid } from 'uuid';
+import { getDummyClient } from '../../testing';
+import { toStreamCollectionName, type StreamType } from '../mongoDBEventStore';
+import {
+  DefaultMongoDBEventStoreCollectionName,
+  mongoDBEventStoreStorage,
+  type MongoDBEventStoreCollectionResolution,
+} from './mongoDBEventStoreStorage';
+
+type TestStreamType = StreamType;
+type OtherTestStreamType = StreamType;
+
+void describe('mongoDBEventStoreStorage', () => {
+  let testStreamTypeName: TestStreamType;
+  let otherStreamTypeName: OtherTestStreamType;
+  let defaultDBName: string;
+  let getConnectedClient: () => Promise<MongoClient>;
+
+  beforeEach(() => {
+    testStreamTypeName = uuid();
+    otherStreamTypeName = uuid();
+    defaultDBName = uuid();
+    getConnectedClient = () =>
+      Promise.resolve(getDummyClient({ defaultDBName }));
```

Comment:
**COMMENT:** I'm using a dummy client, not to mock it, but the storage resolution doesn't require the real MongoDB; it's only based on passing proper parameters to MongoDB API. I can then check if the collection was indeed created, but I'm not trying to mock everything, just stubbing the dummy setup.